accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+| Build id        | The ID of the build.                                                      | run_id           |                                         |                                                                |
+| Build reference | Ref to an external workflow if one is being used, e.g. a trusted builder. | job_workflow_ref | N/A                                     | N/A                                                            |
+
+## Implementation details
+### Sigstore integration in the npm CLI
+Generating build provenance during publish will be opt-in using a command line flag and only run in supported CI/CD systems.
+
+When enabled on publish, the npm CLI will interact with Sigstore through a reverse proxy on npmjs.org: certifying the authenticity of the CD/CD workflow identity token ([fulcio.sigstore.dev](https://docs.sigstore.dev/fulcio/overview/)) and verifying that signing happened during the time the id certificate was valid ([rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview/)).
+
+The signature will be placed on the public Rekor ledger for transparency but a detached copy will be uploaded to the npm registry and used for verification. The detached copy includes proof that it was verified and included by Rekor so we don't need to query it again during verification.
+
+Signature verification will happen transparently during `npm install` and `npm audit signatures` so needs to meet strict performance and reliability uptime requirements as it will be performed for every single npm install. The provenance attestations/signatures will be fetched in bulk from the registry, similar to how advisories are fetched in bulk for all dependencies in the current install. This means verification will not rely on any Sigstore infrastructure to work.
+
+The signing and verification logic will be implemented in a [sigstore-js](https://github.com/sigstore/sigstore-js) client library using Node’s crypto library. Similar clients exist in [Go](https://github.com/sigstore/cosign), [Python](https://github.com/sigstore/sigstore-python) and [Java](http://github.com/sigstore/sigstore-java) and we plan to follow their lead on implementation details.
+
+The npm CLI will rely on the Sigstore JS client library to generate the provenance attestation on publish and to verify the detached signature bundle that's downloaded from the registry.
+
+The Sigstore client library will include direct vendor support for requesting an OIDC id token from supported CI/CD systems, similar to the existing client libraries. It will also include all the required functionality to interact with Fulcio and Rekor, as well as verify the [Sigstore root keys](https://github.com/sigstore/root-signing) using a [TUF](https://theupdateframework.github.io/) client.
+
+#### Fetching Sigstore hosts from npmjs.org
+The npm CLI will fetch Sigstore config from npmjs.org that contains the public Fulcio and Rekor hosts by default:
```

Comment:
Is this something that will be configurable? Alternatively, if I intentionally don't allow the npm CLI to access npmjs.org, can my registry provide this?

Reply:
Yes the idea is that the CLI attempts to fetch this config from the configured registry, so does not need to be npmjs.org. Will clarify 👍